packages/eventstream-handler-node/src/EventStreamPayloadHandler.ts

```diff
@@ -37,10 +38,12 @@ export interface EventStreamPayloadHandlerOptions {
 export class EventStreamPayloadHandler implements IEventStreamPayloadHandler {
   private readonly messageSigner: Provider<MessageSigner>;
   private readonly eventStreamCodec: EventStreamCodec;
+  private readonly systemClockOffsetProvider: Provider<number>;
 
   constructor(options: EventStreamPayloadHandlerOptions) {
     this.messageSigner = options.messageSigner;
     this.eventStreamCodec = new EventStreamCodec(options.utf8Encoder, options.utf8Decoder);
+    this.systemClockOffsetProvider = async () => options.systemClockOffset ?? 0;
```

Comment:
the offset is converted to a provider because the offset must be read at the time of signing and not at the time of initialization.

the offset may have been corrected at some point in the client lifecycle, necessitating a live check of the number value.

Reply:
this won't work immediately because during config resolution we keep copying the object for whatever reason